torch/_inductor/utils.py

```diff
@@ -576,6 +576,17 @@ def free_symbol_has(index: sympy.Expr, pattern: str):
     return any(pattern in v.name for v in index.free_symbols)
 
 
+def is_symbolic(a: Any) -> bool:
+    return isinstance(a, torch.SymInt) or (
+        isinstance(a, torch.Tensor)
+        and any(is_symbolic(x) for x in itertools.chain(a.size(), a.stride()))
+    )
```

Comment:
Why not `has_free_symbols` from symbolic_shapes?

Reply:
Can use that - this is just a refactoring of existing code